waterdata/location_utils.py

```diff
@@ -0,0 +1,148 @@
+            country_lookup = country_state_county_lookups.get(country_code, {})
+            state_lookup = country_lookup.get('state_cd', {}).get(state_code, {})
+            county_lookup = state_lookup.get('county_cd', {})
+
+            transformed_value = county_lookup.get(county_code, {}).get('name', county_code)
+
+        elif key in code_lookups:
+            transformed_value = code_lookups.get(key).get(value, {}).get('name', value)
+
+        else:
+            transformed_value = value
+        transformed_location[key] = transformed_value
+
+    return transformed_location
+
+
+def get_capabilities(location_parameter_records):
+    """
+    Determine the parameters measured at the site.
+
+    :param iterable location_parameter_records: an iterable containing of location parameters
+    :return: USGS parameter codes measured at a site
+    :rtype: set
+
+    """
+    supported_params = set([parameter_record['parm_cd'] for parameter_record in location_parameter_records])
+    return supported_params
+
+
+def get_site_parameter(location_parameter_records, parameter_cd):
+    """
+    Determine the period of record and number of records for a
+    parameter being measured at this site. If the parameter code
+    value specified is not available at this site, None is returned.
+
+    :param iterable location_parameter_records: an iterable containing of location parameter
+    :param str parameter_cd: the USGS parameter code of interest
+    :return: a parameter's "start_date", "end_date", and "record_count" if available; dates are Python date objects
+    :rtype: collection.namedtuple or None
+
+    """
+    try:
+        param_series = next((parameter_record for parameter_record in location_parameter_records
+                             if parameter_record['parm_cd'] == parameter_cd))
+    except StopIteration:
+        return None
+    else:
+        record_start_date = param_series['begin_date']
+        record_end_date = param_series['end_date']
+        start_date = datetime.datetime.strptime(record_start_date, '%Y-%m-%d').date()
+        end_date = datetime.datetime.strptime(record_end_date, '%Y-%m-%d').date()
+        record_count = param_series['count_nu']
+        parameter = Parameter(parameter_cd=parameter_cd,
+                              start_date=start_date,
+                              end_date=end_date,
+                              record_count=record_count
+                              )
+    return parameter
+
+
+def build_linked_data(location_number, location_name, agency_code, latitude, longitude, location_capabilities):
+    """
+    Given site metadata, construct a dictionary / json-ld for the site.
+    The constructed json-ld conforms to the context documents
+    at https://opengeospatial.github.io/ELFIE/json-ld/elf-index.jsonld
+    and https://opengeospatial.github.io/ELFIE/json-ld/hyf.jsonld.
+
+    If a site is missing a piece of data that is needed by the json-ld,
```

Comment:
I find this statement misleading. An empty dictionary is never returned.

Reply:
Yeah. That wasn't updated when the class was removed.